docs/content/concepts/cryptography/nautilus/using-nautilus.mdx

```diff
@@ -0,0 +1,399 @@
+$ sui client faucet
+$ sui client gas
+# deploy the enclave package
+$ cd move/enclave
+$ sui move build
+$ sui client publish
+
+# record ENCLAVE_PACKAGE_ID as env var from publish output
+$ ENCLAVE_PACKAGE_ID=0x14e8b4d8b28ee9aa5ea604f3f33969b3d0f03247b51837f27e17bcf875d3582c
+
+# deploy your dapp logic
+$ cd ../app
+$ sui move build
+$ sui client publish
+
+# record CAP_OBJECT_ID (owned object of type Cap), ENCLAVE_CONFIG_OBJECT_ID (shared object), EXAMPLES_PACKAGE_ID (package containing weather module) as env var from publish output
+
+$ CAP_OBJECT_ID=0xb157d241cc00b7a9b8b0f11d0b4c3e11d8334be95f7e50240962611bd802abff
+$ ENCLAVE_CONFIG_OBJECT_ID=0x58a6a284aaea8c8e71151e4ae0de2350ae877f0bd94adc2b2d0266cf23b6b41d
+$ EXAMPLES_PACKAGE_ID=0x7e712fd9e5e57d87137440cfea77dc7970575a5c3229d78bb7176ab984d94adf
+
+# record the deployed enclave url, e.g. http://<PUBLIC_IP>:3000
+$ ENCLAVE_URL=<DEPLOYED_URL>
+
+# the module name and otw name used to create the dapp, defined in your Move code `fun init`
+$ MODULE_NAME=weather
+$ OTW_NAME=WEATHER
+
+# make sure all env vars are populated
+$ echo $EXAMPLES_PACKAGE_ID
+$ echo $ENCLAVE_PACKAGE_ID
+$ echo $CAP_OBJECT_ID
+$ echo $ENCLAVE_CONFIG_OBJECT_ID
+$ echo 0x$PCR0
+$ echo 0x$PCR1
+$ echo 0x$PCR2
+$ echo $MODULE_NAME
+$ echo $OTW_NAME
+$ echo $ENCLAVE_URL
+
+# =======
+# the two steps that follow (update pcrs, register enclave) can be reused if enclave server is updated
+# =======
+
+# this calls the update_pcrs onchain with the enclave cap and built PCRs, this can be reused to update PCRs if Rust server code is updated
+$ sui client call --function update_pcrs --module enclave --package $ENCLAVE_PACKAGE_ID --type-args "$EXAMPLES_PACKAGE_ID::$MODULE_NAME::$OTW_NAME" --args $ENCLAVE_CONFIG_OBJECT_ID $CAP_OBJECT_ID 0x$PCR0 0x$PCR1 0x$PCR2
+
+# optional, give it a name you like
+$ sui client call --function update_name --module enclave --package $ENCLAVE_PACKAGE_ID --type-args "$EXAMPLES_PACKAGE_ID::$MODULE_NAME::$OTW_NAME" --args $ENCLAVE_CONFIG_OBJECT_ID $CAP_OBJECT_ID "weather enclave, updated 2025-05-13"
+
+# this script calls the get_attestation endpoint from your enclave url and use it to calls register_enclave onchain to register the public key, results in the created enclave object
+$ sh ../../register_enclave.sh $ENCLAVE_PACKAGE_ID $EXAMPLES_PACKAGE_ID $ENCLAVE_CONFIG_OBJECT_ID $ENCLAVE_URL $MODULE_NAME $OTW_NAME
+
+# record the created shared object ENCLAVE_OBJECT_ID as env var from register output
+$ ENCLAVE_OBJECT_ID=0xe0e70df5347560a1b43e5954267cadd1386a562095cb4285f2581bf2974c838d
+```
+
+You can view an example of an enclave config object containing PCRs [on SuiScan](https://testnet.suivision.xyz/object/0x58a6a284aaea8c8e71151e4ae0de2350ae877f0bd94adc2b2d0266cf23b6b41d). You can also view an [example of an enclave object](https://testnet.suivision.xyz/object/0xe0e70df5347560a1b43e5954267cadd1386a562095cb4285f2581bf2974c838d) containing the enclave public key.
```

Comment:
I forgot to point out that we don't want to provide links to testnet data as there's no guarantee this info won't get wiped. We can address later but something to keep in mind on future writings.

Reply:
That's a good point, especially with the discussions on a potential testnet wipe. cc @joyqvq